### PR TITLE
M2 linreg

### PR DESCRIPTION
**Pull request by cmmclaug**

### CODE REVIEW

File: aridanalysis/aridanalysis.py [hunk 1]
```diff
@@ -44,23 +56,84 @@ def arid_linreg(data_frame, response, features=[], estimator=None, regularizatio
...
+                                                    alpha = alpha/3)
+    elif regularization == "L1L2":
+        skl_model = ElasticNet(alpha, fit_intercept = False).fit(X, y)
+        sm_model = sm.OLS(y, X).fit_regularized(L1_wt = 0.5, 
+                                                alpha = alpha)
+    else:
+        skl_model = LinearRegression(fit_intercept = False).fit(X, y)
+        sm_model = sm.OLS(y, X).fit()
+     
+    # Display model coefficients to user
```
Comment (nphaterp) [inline aridanalysis/aridanalysis.py:130]: we need to make sure that this is consistent with the other functions

File: aridanalysis/aridanalysis.py [hunk 2]
```diff
@@ -30,11 +42,11 @@ def arid_eda(data_frame, response, features=[]):
     """
     return None
 
-def arid_linreg(data_frame, response, features=[], estimator=None, regularization=None):
+def arid_linreg(df, response, features=[], regularization=None, alpha=1):
```
Comment (nphaterp) [inline aridanalysis/aridanalysis.py:45]: we need to make sure this change is reflected in the other functions (data_frame -> df )

File: tests/test_aridanalysis.py [hunk 3]
```diff
@@ -1,5 +1,123 @@
...
+
+import sys, os
+myPath = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, myPath + '/../aridanalysis')
+import error_strings as errors
 
 def test_version():
-    assert __version__ == '0.1.0'
\ No newline at end of file
+    assert __version__ == '0.1.0'
```
Comment (nphaterp) [inline tests/test_aridanalysis.py:18]: Don't really know what this is? Might need to adjust based on the current version number

File: aridanalysis/aridanalysis.py [hunk 4]
```diff
@@ -44,23 +56,84 @@ def arid_linreg(data_frame, response, features=[], estimator=None, regularizatio
...
+    statsmodels.regression.linear_model
+        A fitted statsmodel configured with the chosen input parameters
 
     Examples
     --------
     >>> from aridanalysis import aridanalysis
     >>> aridanalysis.arid_linreg(df, income)
     """
-    return None
+def arid_linreg(df, response, features=[], regularization=None, alpha=1):
```
Comment (nphaterp) [inline aridanalysis/aridanalysis.py]: this line is duplicated ?

**Review by nphaterp — CHANGES_REQUESTED** [review]

File: aridanalysis/aridanalysis.py (hunk 1 above)
Comment (cmmclaug) [reply aridanalysis/aridanalysis.py:130]: Agreed, there's an issue with the non-OLS models not outputting summaries...Will have to figure out a way to standardize.

File: aridanalysis/aridanalysis.py (hunk 2 above)
Comment (cmmclaug) [reply aridanalysis/aridanalysis.py:45]: Agreed, I like the df usage change though.

File: tests/test_aridanalysis.py (hunk 3 above)
Comment (cmmclaug) [reply tests/test_aridanalysis.py:18]: Yes, will look up. I don't know if we'd have to manually do this though?

File: aridanalysis/aridanalysis.py (hunk 2 above)
Comment (nphaterp) [reply aridanalysis/aridanalysis.py:45]: Ok, we can proceed with that

File: aridanalysis/aridanalysis.py (hunk 4 above)
Comment (cmmclaug) [reply aridanalysis/aridanalysis.py]: I will fix!

**Review by nphaterp — APPROVED** [review]
LGTM, thanks for making those changes

**Pull request merged**